src/thor/astar_bss.cc

```diff
@@ -149,8 +149,10 @@ void AStarBSSAlgorithm::ExpandForward(GraphReader& graphreader,
     // directed edge), if no access is allowed to this edge (based on costing method),
     // or if a complex restriction exists.
     uint8_t has_time_restrictions = -1;
+    bool is_dest = destinations_.find(edgeid) != destinations_.cend();
```

Comment:
what's perf penalty with additional map lookups added (here and the rest)?

Reply:
I answered about performance below, I have tested using other tools. 
Hashtable lookups is O(1). 
You can think that next operations will be performed:
```
bucket = edgeid % hashtable_size
list = array[bucket]
return list.find(edgeid)
```
Size of lists -> 1 in good hastables.

Reply:
the lookup is amortized constant time yes but that doesnt mean its free. it costs clock cycles and its in the tight loop of the algorithm. how did you validate that performance was unimpacted? can you share some numbers (especially for bidirection_astar)?

Reply:
```
// master
time ./run_with_server.py --test-file ../test_requests/de_benchmark_routes.txt --url http://localhost:8002/route --concurrency 1
Placing 2550 results in 20210503_135856_de_benchmark_routes
9%
14%
18%
24%
29%
34%
39%
44%
49%
55%
60%
65%
70%
75%
80%
85%
90%
95%
100%
./run_with_server.py --test-file ../test_requests/de_benchmark_routes.txt      4.08s user 0.91s system 5% cpu 1:39.65 total
// this PR
time ./run_with_server.py --test-file ../test_requests/de_benchmark_routes.txt --url http://localhost:8002/route --concurrency 1
Placing 2550 results in 20210503_140129_de_benchmark_routes
9%
14%
18%
24%
29%
34%
39%
44%
50%
55%
60%
65%
70%
75%
80%
85%
90%
95%
100%
./run_with_server.py --test-file ../test_requests/de_benchmark_routes.txt      4.09s user 0.91s system 5% cpu 1:39.65 total
```